core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -239,6 +239,10 @@ abstract class RawModule extends BaseModule {
       case (false, false)                               =>
         // For non-probe, directly create Nodes for lhs, skipping visibility check to support BoringUtils.drive.
         (left, right) match {
+          case (lhsOpaque: OpaqueType, rhsOpaque: OpaqueType)
+              if DataMirror
+                .isProperty(lhsOpaque.allElements.head) && DataMirror.isProperty(rhsOpaque.allElements.head) =>
+            PropAssign(si, Node(lhsOpaque.allElements.head), Node(rhsOpaque.allElements.head))
```

Comment:
```suggestion
          case (lhsOpaque: Record, rhsOpaque: Record)
              if lhsOpaque._isOpaqueType && rhsOpaque._isOpaqueType && DataMirror.isProperty=>
            secretConnection(lhsOpaque, rhsOpaque)
```
`OpaqueType` the trait isn't enough to check (and note the trait is self-typed by `Record`), instead we match on `Record` and check `_isOpaqueType`. Also this needs to be recursive in some way because you can have `OpaqueTypes` of `OpaqueTypes`. It's turtles all the way down!

Reply:
Ah good point. What if DataMirror.isProperty handles _isOpaqueType and recursively checking? Then we could have one case here that just uses the `if DataMirror.isProperty` condition?

Reply:
Ugh I guess it's not that easy, this needs to recursively pick off the allElements.head till it gets to the bottom.

Reply:
I think [fa90b8f](https://github.com/chipsalliance/chisel/pull/4337/commits/fa90b8f5237aa9cf4cdd99700ecc4858ac7300ea) should handle it, this is now using _isOpaqueType and recursively unwrapping OpaqueTypes.